Fix invalid escape sequences in ECs

Flamefire opened the pull request:
Similar to https://github.com/easybuilders/easybuild-easyconfigs/pull/11148 this handles the invalid escape sequences by using raw strings for all ECs

I.e. it fixes the `<string>:25: DeprecationWarning: invalid escape sequence \$` warnings when parsing an EC and the related flake8 warning

This is clean with `flake8 --select W605  --filename '*.eb' -j5` now

- [x] Requires https://github.com/easybuilders/easybuild-easyconfigs/pull/17279

lexming reviewed (commented):
I think that this change will cause issues with [`quote_str()`](https://github.com/easybuilders/easybuild-framework/blob/b74b830e61920b1c04f6e389ccd567197aaf8426/easybuild/tools/utilities.py#L57), which can escape backslashes. Raw strings will end up having double backslashes.

Flamefire commented:
If that is the case maybe the function needs a fix? I'm not sure how it is used though so can't tell if it causes problems
If so then better use escaped backslash

Flamefire commented:
Quick ping here. Still getting those DeprecationWarnings and I think we should fix those before they become errors. And the problem just get worse as more and more ECs get added by copying those "mistakes"

Flamefire commented:
Rebased again. What is the status here @boegel ?

jfgrimm reviewed (commented):
As decided in issue #16330, we have deprecated the use of `True` to signify a system-toolchain dependency (#16384), in favour of the more intuitive `SYSTEM` template constant. Due to the change in the test suite, please run `eb --sync-pr-with-develop 11149` and update the PR to use `SYSTEM` instead.

Flamefire commented:
@jfgrimm I'm not sure your comment applies here as this is only a refactoring to fix the style/warnings not changing or introducing anything

Anyway can this and #11148 be reviewed and merged please so we can enable flake8 on CI and avoid reintroducing the issues I already fixed?

I just rebased this, resolved the conflict and checked that `flake8 --select W605  --filename '*.eb' -j5` doesn't return anything

jfgrimm commented on easybuild/easyconfigs/s/shovill/shovill-1.1.0-gompi-2021b.eb:
```diff
@@ -33,7 +33,7 @@ dependencies = [
 
 postinstallcmds = [
-    """echo exec java -jar \$EBROOTTRIMMOMATIC/trimmomatic-*.jar '"$@"' > %(installdir)s/bin/trimmomatic""",
+    """echo exec java -jar \\$EBROOTTRIMMOMATIC/trimmomatic-*.jar '"$@"' > %(installdir)s/bin/trimmomatic""",
```
consistency across the `shovill` easyconfigs
```suggestion
    r"""echo exec java -jar \$EBROOTTRIMMOMATIC/trimmomatic-*.jar '"$@"' > %(installdir)s/bin/trimmomatic""",
```

jfgrimm commented on easybuild/easyconfigs/s/SciPy-bundle/SciPy-bundle-2021.05-gomkl-2021a.eb:
```diff
@@ -55,7 +55,7 @@ exts_list = [
     }),
     ('pandas', '1.2.4', {
-        'preinstallopts': """sed -i 's@extra_compile_args = \["-Werror"\]@extra_compile_args = []@g' setup.py && """,
+        'preinstallopts': r"""sed -i 's@extra_compile_args = \["-Werror"\]@extra_compile_args = []@g' setup.py && """,
```
consistency across `SciPy-bundle` easyconfigs
```suggestion
        'preinstallopts': """sed -i 's@extra_compile_args = \\["-Werror"\\]@extra_compile_args = []@g' setup.py && """,
```

jfgrimm commented on easybuild/easyconfigs/m/mosdepth/mosdepth-0.2.4-foss-2018b.eb:
```diff
@@ -31,7 +31,7 @@ dependencies = [
 
 install_cmd = "cd %(builddir)s/hts-nim-*/ && nimble install --nimbleDir:%(installdir)s --verbose -y && "
-install_cmd += "cd ../mosdepth-*/ && sed -i 's/0\.2\.2/%(version)s/g' mosdepth.nimble && "
+install_cmd += r"cd ../mosdepth-*/ && sed -i 's/0\.2\.2/%(version)s/g' mosdepth.nimble && "
```
consistency across `mosdepth` easyconfigs
```suggestion
install_cmd += "cd ../mosdepth-*/ && "
install_cmd += r"sed -i 's/0\.2\.2/%(version)s/g' mosdepth.nimble && "
```

jfgrimm commented on easybuild/easyconfigs/g/GDAL/GDAL-2.4.4-intel-2021b.eb:
```diff
@@ -49,7 +49,7 @@ dependencies = [
 ]
 
-preconfigopts = "sed -e 's/-llapack/\$LIBLAPACK/g' -i.eb configure && "
+preconfigopts = "sed -e 's/-llapack/$LIBLAPACK/g' -i.eb configure && "
```
consistency across `GDAL` easyconfigs
```suggestion
preconfigopts = r"sed -e 's/-llapack/\$LIBLAPACK/g' -i.eb configure && "
```

jfgrimm commented on easybuild/easyconfigs/g/GDAL/GDAL-2.4.4-foss-2021b.eb:
```diff
@@ -49,7 +49,7 @@ dependencies = [
 ]
 
-preconfigopts = "sed -e 's/-llapack/\$LIBLAPACK/g' -i.eb configure && "
+preconfigopts = "sed -e 's/-llapack/$LIBLAPACK/g' -i.eb configure && "
```
consistency across `GDAL` easyconfigs
```suggestion
preconfigopts = r"sed -e 's/-llapack/\$LIBLAPACK/g' -i.eb configure && "
```

jfgrimm commented on easybuild/easyconfigs/c/CGAL/CGAL-4.14.3-iimpi-2021a.eb:
```diff
@@ -28,7 +28,7 @@ dependencies = [
 ]
 
-configopts = "-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "
+configopts = '-DOPENGL_INCLUDE_DIR="$EBROOTMESA/include;$EBROOTLIBGLU/include" '
```
consistency across `CGAL` easyconfigs
```suggestion
configopts = r"-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "
```

jfgrimm commented on easybuild/easyconfigs/c/CGAL/CGAL-4.14.3-iimpi-2020a-Python-3.8.2.eb:
```diff
@@ -29,7 +29,7 @@ dependencies = [
 ]
 
-configopts = "-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "
+configopts = '-DOPENGL_INCLUDE_DIR="$EBROOTMESA/include;$EBROOTLIBGLU/include" '
```
consistency across `CGAL` easyconfigs
```suggestion
configopts = r"-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "
```

jfgrimm commented on easybuild/easyconfigs/c/CGAL/CGAL-4.14.3-gompi-2021b.eb:
```diff
@@ -28,7 +28,7 @@ dependencies = [
 ]
 
-configopts = "-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "
+configopts = '-DOPENGL_INCLUDE_DIR="$EBROOTMESA/include;$EBROOTLIBGLU/include" '
```
consistency across `CGAL` easyconfigs
```suggestion
configopts = r"-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "
```

jfgrimm commented on easybuild/easyconfigs/c/CGAL/CGAL-4.14.3-gompi-2021a.eb:
```diff
@@ -28,7 +28,7 @@ dependencies = [
 ]
 
-configopts = "-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "
+configopts = '-DOPENGL_INCLUDE_DIR="$EBROOTMESA/include;$EBROOTLIBGLU/include" '
```
consistency across `CGAL` easyconfigs
```suggestion
configopts = r"-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "
```

Flamefire replied:
There is no need for the escape when quoting the value. This is actually correct as the original (unquoted) would break if `$EBROOTMESA` contains spaces

jfgrimm replied:
in which case, most of the other `CGAL` easyconfigs should be updated since they use `r"-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include\;$EBROOTLIBGLU/include "`

Flamefire replied:
True. Done

Flamefire commented:
I extracted the shovill changes into a [separate PR](https://github.com/easybuilders/easybuild-easyconfigs/pull/17279) and added a CI check to avoid reintroducing this. That can be extended once the other flake8-related PRs have been merged.

lexming commented:
Test report by @lexming
**SUCCESS**
Build succeeded for 58 out of 58 (58 easyconfigs in total)
node300.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 3.6.8
See https://gist.github.com/lexming/203867d0566011912c771172b4ebea10 for a full test report.

lexming reviewed (approved):
LGTM

lexming commented:
Going in, thanks @Flamefire !

lexming commented:
Tests rolled-back to normal state in https://github.com/easybuilders/easybuild-easyconfigs/pull/17304
